xmake/rules/c++/modules/modules_support/clang.lua

```diff
@@ -0,0 +1,272 @@
+    end
+    assert(implicitmodulesflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -fimplicit-module-maps" .. os.tmpdir(), "cxxflags", {flagskey = "clang_implicit_module_path"}) then
+        implicitmodulemapsflag = "-fimplicit-module-maps"
+    end
+    assert(implicitmodulemapsflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -fprebuilt-module-path=" .. os.tmpdir(), "cxxflags", {flagskey = "clang_prebuild_module_path"}) then
+        prebuiltmodulepathflag = "-fprebuilt-module-path="
+    end
+    assert(prebuiltmodulepathflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -fmodules-cache-path=" .. os.tmpdir(), "cxxflags", {flagskey = "clang_modules_cache_path"}) then
+        modulecachepathflag = "-fmodules-cache-path="
+    end
+    assert(modulecachepathflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -emit-module", "cxxflags", {flagskey = "clang_emit_module"}) then
+        emitmoduleflag = " -emit-module"
+    end
+    assert(emitmoduleflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -fmodule-file=" .. os.tmpfile() .. get_bmi_ext(), "cxxflags", {flagskey = "clang_module_file"}) then
+        modulefileflag = "-fmodule-file="
+    end
+    assert(modulefileflag, "compiler(clang): does not support c++ module!")
+
+    if compinst:has_flags((modulesflag or modulestsflag) .. " -emit-module-interface", "cxxflags", {flagskey = "clang_emit_module_interface"}) then
+        emitmoduleinterfaceflag = "-emit-module-interface"
+    end
+    assert(emitmoduleinterfaceflag, "compiler(clang): does not support c++ module!")
+
+    import("common")
+    local cachedir = common.get_cache_dir(target)
+    local stlcachedir = common.get_stlcache_dir(target)
+
+    -- add module flags
+    target:add("cxxflags", modulesflag or modulestsflag)
+
+    -- add the module cache directory
+    target:add("cxxflags", implicitmodulesflag, {force = true})
+    target:add("cxxflags", implicitmodulemapsflag, {force = true})
+
+    target:add("cxxflags", prebuiltmodulepathflag .. cachedir, prebuiltmodulepathflag .. stlcachedir, {force = true})
+end
+
+function toolchain_include_directories(target)
+    if is_plat("linux") then
+        return { "/usr/include/**", "/usr/local/include/**" }
+    end
+
+    return {}
+end
+
+function generate_dependencies(target, sourcebatch, opt)
+    import("common")
+    local cachedir = common.get_cache_dir(target)
+
+    for _, sourcefile in ipairs(sourcebatch.sourcefiles) do 
+        local dependfile = target:dependfile(sourcefile)
+        depend.on_changed(function()
+			progress.show(opt.progress, "${color.build.object}generating.cxx.module.deps %s", sourcefile)
```

Comment:
incorrect indent

Reply:
i don't understand where is the incorrect indent

Reply:
we should use `space4` instead of tab chars.

they are two tab chars here.